### PR TITLE
WIP: Attempt to get the server to open the Root.spawnable

### PR DESCRIPTION
**Pull request by AMZN-Gene**
WiP. Trying to get the server to open the Root.spawnable, but not yet working

Signed-off-by: Gene Walters <genewalt@amazon.com>

### CODE REVIEW

File: Code/Framework/AzFramework/AzFramework/Spawnable/InMemorySpawnableAssetContainer.cpp [hunk 1]
```diff
@@ -132,23 +133,26 @@ namespace AzFramework
...
         assetDataInfoContainer.emplace_back(AZStd::make_pair<AZ::Data::AssetData*, AZ::Data::AssetInfo>(assetData, assetInfo));
 
-        return CreateInMemorySpawnableAsset(assetDataInfoContainer, loadReferencedAssets, rootSpawnableName);
+        // Remove the .spawnable file extension before passing it to the other CreateInMemorySpawnableAsset method, which will add it back in
+        AZStd::string spawablePathSansFileExtension = assetRelativePath;
+        if (spawablePathSansFileExtension.ends_with(Spawnable::DotFileExtension))
+        {
+            spawablePathSansFileExtension = assetRelativePath.substr(0, assetRelativePath.length() - strlen(Spawnable::DotFileExtension));
+        }
+        return CreateInMemorySpawnableAsset(assetDataInfoContainer, loadReferencedAssets, spawablePathSansFileExtension);
```
Comment (michabr) [inline Code/Framework/AzFramework/AzFramework/Spawnable/InMemorySpawnableAssetContainer.cpp:155]: AzFramework::StringFunc::Path::HasExtension and AzFramework::StringFunc::Path::StripExtension can be used here instead.

File: Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp [hunk 2]
```diff
@@ -91,6 +96,12 @@ namespace Multiplayer
         {
             // This is the last expected packet
             // Read all assets out of the buffer
+            // Create in-memory spawnables for the level, root.spawnable and root.network.spawnable (if level contains network entities)
+            if (m_inMemorySpawnableAssetContainer != nullptr)
+            {
+                m_inMemorySpawnableAssetContainer->ClearAllInMemorySpawnableAssets();
```
Comment (michabr) [inline Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp:102]: I wonder if we should add this to a destructor.

**Review by michabr — COMMENTED** [review]
Nice progress.

File: Code/Framework/AzFramework/AzFramework/Spawnable/InMemorySpawnableAssetContainer.cpp (hunk 1 above)
Comment (AMZN-Gene) [reply Code/Framework/AzFramework/AzFramework/Spawnable/InMemorySpawnableAssetContainer.cpp]: Fixed! Root.network.spawnable will still have the .network extension. I think that needs to the be stripped too, but will investigate more before the next pull request

File: Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp (hunk 2 above)
Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp:107]: fixed!

**Review by michabr — APPROVED** [review]

**Pull request merged**